File: Gestor Hospitalario/Models/CentroMedico.cs
using System.ComponentModel.DataAnnotations;

namespace Gestor_Hospitalario.Models
{
    public class CentroMedico
    {
        [Key]
        public int CentroID { get; set; }
        public required string Nombre { get; set; }
        public required string Direccion { get; set; }
        public required string Telefono { get; set; }
        public required string Email { get; set; }

        public ICollection<Medico>? Medicos { get; set; }
        public ICollection<Empleado>? Empleados { get; set; }
        public ICollection<ConsultaMedica>? ConsultasMedicas { get; set; }
    }
}

File: gestor-hospitalario-react/src/@types/centroMedico.ts
export interface CentroMedico {
  centroID: number;
  nombre: string;
  direccion: string;
  telefono: string;
  email: string;
}

export interface CentroMedicoCreate {
  nombre: string;
  direccion: string;
  telefono: string;
  email: string;
}

export interface CentroMedicoUpdate extends CentroMedicoCreate {
  centroID: number;
}
